src/ape_geth/query.py

```diff
@@ -0,0 +1,34 @@
+from functools import singledispatchmethod
+from typing import Iterator, Optional
+
+from ape.api import ReceiptAPI
+from ape.api.query import ContractCreationQuery, QueryAPI, QueryType
+from ape.exceptions import QueryEngineError
+from ape_geth.provider import BaseGethProvider
+
+
+class OTSQueryEngine(QueryAPI):
+    @singledispatchmethod
+    def estimate_query(self, query: QueryType) -> Optional[int]:  # type: ignore[override]
+        return None
+
+    @singledispatchmethod
+    def perform_query(self, query: QueryType) -> Iterator:  # type: ignore[override]
+        raise QueryEngineError(
+            f"{self.__class__.__name__} cannot handle {query.__class__.__name__} queries."
+        )
+
+    @estimate_query.register
+    def estimate_contract_creation_query(self, query: ContractCreationQuery) -> Optional[int]:
+        if self.network_manager.active_provider:
+            if not isinstance(self.provider, BaseGethProvider):
+                return None
+
+        # About 225 ms per query
```

Comment:
Etherscan's is 300. I don't know what to make this really but I figured since you are likely running the node yourself, it should be faster than Etherscan's.

When I tried it out, it seemed like the first call to OTS was slower than the rest, I am not sure if it does some caching or what, or if it was a fluke in my own internet because I was using wifi a bit far away at that time.

Reply:
OK after some observation, I realized it is actually slower for me when using a remote connection to an OTS node, so I made it account for that, in a way. If the URL is a remote node, it is slower than `ape-etherscan`. If it is a local node, it is faster.